in_toto/in_toto_sign.py

```diff
@@ -207,33 +234,87 @@ def main():
   arguments it then dumps the corresponding file.
   """
   args = parse_args()
-  rsa_key = in_toto.util.import_rsa_key_from_file(args.key)
+  length = len(args.keys)
+  rsa_key = in_toto.util.import_rsa_key_from_file(args.keys[length-1])
+  FILENAME_FORMAT_LAYOUT = "{file_name}.{keyid:.8}.layout"
+  FILENAME_FORMAT_LINK = "{file_name}.{keyid:.8}.link"
+  index=args.signablepath.rfind('/')
+  source_file_name=args.signablepath[index+1:]
 
   if args.verbose:
     log.logging.getLogger().setLevel(log.logging.INFO)
 
-  try:
-    if args.operator == 'sign':
+  if args.operator == 'sign':
+    try:
       if args.replace_sig:
-        signable_object = replace_sign(args.signablepath, args.key)
-
+        signable_object = replace_sign(args.signablepath, args.keys)
       else:
-        signable_object = add_sign(args.signablepath, args.key)
+        signable_object = add_sign(args.signablepath, args.keys)
 
       if args.infix:
-        signable_object.dump(key=rsa_key)
-        sys.exit(0)
+        if not args.destination:
+          if signable_object._type=='layout':
+            fn = FILENAME_FORMAT_LAYOUT.format(file_name=source_file_name,
+              keyid=rsa_key['keyid'])
+            signable_object.dump(fn)
+            sys.exit(0)
+
+          else:
+            signable_object.dump(key=rsa_key)
+            sys.exit(0)
```

Comment:
> As a general note, please make sure that unittest coverage does not decrease with your code additions.

Have you tried using coverage's `--fail-under`?  This should make it so that if coverage falls below a certain threshold, a build failure occurs.
```
coverage report -m --fail-under 97
```

I wonder if you can give this option a decimal number...